cfme/tests/automate/test_instance.py

```diff
@@ -354,3 +354,117 @@ def test_quota_source_value(request, entity, search, copy_quota_instance, generi
+    """
+    Bugzilla:
+        1441353
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseimportance: high
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.9
+        casecomponent: Automate
+        tags: automate
+    """
+    state1 = fauxfactory.gen_alpha()
+    state2 = fauxfactory.gen_alpha()
+    state3 = fauxfactory.gen_alpha()
+    for state in [state1, state2, state3]:
+        klass.schema.add_fields({'name': state, 'type': 'State'})
+
+    # Adding method to execute via parent instance(state machine instance)
+    method = klass.methods.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        location='inline',
+        script="""
+               \n$evm.log(:info, "Hello from method of parent instance")
+                \nexit MIQ_STOP
+                """
+    )
+    request.addfinalizer(method.delete_if_exists)
+
+    # Adding parent instance to call automate method and child instances in other class
+    instance = klass.instances.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        description=fauxfactory.gen_alphanumeric(),
+        fields={state3: {"value": f"METHOD::{method.name}"}}
+    )
+    request.addfinalizer(instance.delete_if_exists)
+
+    # Creating other class for two child instances
+    klass2 = namespace.classes.create(
+        name=fauxfactory.gen_alpha(),
+        display_name=fauxfactory.gen_alpha(),
+        description=fauxfactory.gen_alpha()
+    )
+    request.addfinalizer(klass2.delete_if_exists)
+
+    for state in [state1, state2]:
+        klass2.schema.add_fields({'name': state, 'type': 'State'})
+
+    # Creating three child methods to execute via child instances
+    child_method = list(map(lambda num: klass2.methods.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        location='inline',
+        script=f"""
+               \n$evm.log(:info, "This is method {num}")
+                \nexit {process.upper()}
+                """
+    ), ["first", "second", "third"]))
+
+    fields = [{state1: {'value': f"METHOD::{child_method[0].name}"},
+              state2: {'value': f"METHOD::{child_method[1].name}"}},
+             {state1: {'value': f"METHOD::{child_method[2].name}"}}]
+
+    # Creating two child instances
+    child_inst = list(map(lambda field: klass2.instances.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        description=fauxfactory.gen_alphanumeric(),
+        fields=field
+    ), fields))
+
+    # Updating parent instance fields to execute child instances
+    with update(instance):
+        instance.fields = {
+            state1: {
+                "value": f"/{domain.name}/{namespace.name}/{klass2.name}/{child_inst[0].name}"
```

Comment:
How do you feel about using `"/".join(child_inst[0].tree_path_name_only[1:])` instead of this long string formating?

Reply:
I would like to keep it as it is.
Because I find it more readable than using ```tree_path_name_only``` otherwise it will be very difficult for me to understand from which domain/namespace/class method is getting execute.